### PR TITLE
hwdb: 60-sensor.hwdb: Add proximity sensor udev property

### PR DESCRIPTION
**Pull request by agx**
The match is currently on the sensor chip only since there's no DMI available.

I've marked this as 'RFC' since there's currently no match on the DMI part due to lack of DMI on this arm64 platform.

### CODE REVIEW

**Comment by hadess** [comment]
You'll need to expand `hwdb.d/parse_hwdb.py` to recognise the option you're adding.

**Comment by hadess** [comment]
My guess, for device tree devices, would that this level information should be in the DT, and the kernel driver (or rather the IIO subsystem) should export it to user-space as a device property. iio-sensor-proxy could then pick it up from `/sys`.

@jwrdegoede what do you think?

**Comment by agx** [comment]
@hadess i've been thinking the same regarding DT initially but from what i read on the iio list they're trying to avoid that.

**Comment by agx** [comment]
It seems i missed that the mount matrix can be set via DT, so i brought this over to the iio kernel list as well for feedback: https://lore.kernel.org/linux-iio/20200210154153.GA26903@bogon.m.sigxcpu.org/T/#u

**Comment by jwrdegoede** [comment]
I agree with @hadess for platforms with DMI this info really belongs in device-tree. We wish that on X86 the accel-orientation was in some ACPI table too instead of having to add all these quirks, and for some sensors we do get the info from ACPI using some vendor specific tables.

Adding something like this, without being able to match on a specific model is not acceptable IMHO. So you need to either add some way to identify the model, or just put it in device-tree, with just putting it in device-tree being strongly preferred.

**Comment by agx** [comment]
> Adding something like this, without being able to match on a specific model is not acceptable IMHO. So you need to either add some way to identify the model, or just put it in device-tree, with just putting it in device-tree being strongly preferred.

see https://lore.kernel.org/linux-iio/cover.1581947007.git.agx@sigxcpu.org/

File: hwdb.d/60-sensor.hwdb [hunk 1]
```diff
@@ -529,6 +533,12 @@ sensor:modalias:acpi:KIOX000A*:dmi:*:rvnPOV:rnI102A:*
 sensor:modalias:i2c:bmc150_accel:dmi:bvnINSYDECorp.:*:svnInsyde:pnBayTrail:*:rvn105B:rn0E57:*
  ACCEL_MOUNT_MATRIX=1, 0, 0; 0, -1, 0; 0, 0, 1
 
+#########################################
+# Purism
+#########################################
+sensor:modalias:of:NproxT*Cvishay,vcnl4040Cvishay,vcnl4200:*
+ PROXIMITY_NEAR_LEVEL=220
```
Comment (hadess) [inline hwdb.d/60-sensor.hwdb]: Can you split the portion of the patch that relates to the documentation additions from adding the device?

**Comment by hadess** [comment]
I'm fine with merging this patchset without the Purism device quirk added.

File: hwdb.d/60-sensor.hwdb (hunk 1 above)
Comment (agx) [reply hwdb.d/60-sensor.hwdb]: yes, i dropped the device completely as per above discussion

**Pull request merged**